lxd/storage_btrfs.go

```diff
@@ -2802,7 +2802,7 @@ func (s *storageBtrfs) MigrationSink(conn *websocket.Conn, op *operation, args M
 	}
 
 	containersMntPoint := getContainerMountPoint(args.Container.Project(), s.pool.Name, "")
-	err := createContainerMountpoint(containersMntPoint, args.Container.Path(), args.Container.IsPrivileged())
+	err := createMountpoint(containersMntPoint, args.Container.Path(), args.Container.IsPrivileged(), true)
```

Comment:
Any idea what this is trying to achieve?

From what I understand, this is creating a symlink from the container's path to the storage-pool's containers path, which seems very odd.

Reply:
I have no idea, it's very odd indeed, but for the purpose of fixing this bug I wanted to keep the change at minimum and not change existing behavior that I was not sure about.

This PR keeps the semantics of ```createContainerMountpoint``` totally unchanged. So all call sites that "legitimately" use it won't get any surprise.

However the one use of ```createContainerMountpoint``` in this particular spot does not indeed seem to be legitimate: for this case I minimized the change by only running os.Chmod if the directory is created from scratch.

Now, in practice I think that the os.Chmod code path will never be taken, because at least in case of btrfs the ```containers``` directory is created upfront so it should always be there. I also think that if the code path was actually taken (because the directory doesn't exist), that os.Chmod would probably be wrong, but it would match existing behavior.

If you feel confident that we can get rid of this mess altogether by simply dropping this call site (because we can safely assume the containers directory is there and it has the correct permission), I can change the PR accordingly.

Reply:
Yeah, this call looks completely bogus to me, if it's there for a reason, we have a much deeper issue that we should fix. I'd say, get rid of that weird createContainerMountpoint call, which likely means the rest of this PR can go away. Then lets see if Jenkins is happy with it. We have reasonably good migration testing in place so if it's actually needed for some reason, it should be pretty obvious.

Reply:
Done. Let's see what jenkins says.

Reply:
It seems that the call is actually needed, for some reason:

```
Error: Failed container creation: Error transferring container data: stat /lxc-ci/build/tmp.uXGg6Ztljd/go/src/github.com/lxc/lxd/test/tmp.q6Z/ztT/storage-pools/lxdtest-ztT/containers/proj_: no such file or directory
```

see the full [job output](https://jenkins.linuxcontainers.org/job/lxd-github-pull-test/6020/arch=amd64,backend=btrfs,compiler=golang-1.12,restrict=lxd-priv/console#console-section-3).

Reply:
That's one odd path to be checking, also means that if this is indeed used and two containers are being migrated at the same time, `bad things` will happen.